src/localstorage/localstorage.ts

```diff
@@ -23,9 +23,20 @@ function getLastTaskId() {
   return lastSavedId;
 }
 
+function updateTaskInLocalStorage(task: Task) {
+  const previosStoredTasks = getTasksFromLocalStorage();
+  const filteredPrevTasks = previosStoredTasks.filter(
+    (item: Task) => Number(item.id) !== Number(task.id)
+  );
+  const updatedTasks = [...filteredPrevTasks, task];
+  localStorage.removeItem("tasks");
+  localStorage.setItem("tasks", JSON.stringify(updatedTasks));
```

Comment:
You don't need to remove tasks prior to setting it. Setting it should automatically replace the current array.